sql/src/main/java/io/crate/operation/scalar/string/LowerFunction.java

```diff
@@ -0,0 +1,96 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.operation.scalar.string;
+
+import com.google.common.collect.ImmutableList;
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.FunctionInfo;
+import io.crate.metadata.Scalar;
+import io.crate.operation.Input;
+import io.crate.operation.scalar.ScalarFunctionModule;
+import io.crate.planner.symbol.Function;
+import io.crate.planner.symbol.Symbol;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.lucene.BytesRefs;
+
+import java.util.List;
+import java.util.Locale;
+
+public class LowerFunction extends Scalar<BytesRef, Object> {
+    public static final String NAME = "lower";
+
+    public static void register(ScalarFunctionModule module) {
+        List<DataType> supportedLowerTypes = ImmutableList.<DataType>of(
+                DataTypes.STRING);
+
+        for (DataType dataType : supportedLowerTypes) {
+            // without locale
+            module.register(new LowerFunction(new FunctionInfo(
+                    new FunctionIdent(NAME, ImmutableList.of(dataType)),
+                    DataTypes.STRING)
+            ));
+            // with locale
+            module.register(new LowerFunction(new FunctionInfo(
+                    new FunctionIdent(NAME, ImmutableList.of(dataType, DataTypes.STRING)),
+                    DataTypes.STRING)
+            ));
+        }
+    }
+
+    private FunctionInfo info;
+
+    public LowerFunction(FunctionInfo info) {
+        this.info = info;
+    }
+
+    @Override
+    public BytesRef evaluate(Input<Object>... args) {
+        if (hasNullInputs(args)) {
+            return null;
+        }
+
+        BytesRef inputByteRef = BytesRefs.toBytesRef(args[0].value());
+
+        Locale currentLocale = Locale.getDefault();
+        if (args.length == 2) {
+            String localeString = BytesRefs.toBytesRef(args[1].value()).utf8ToString();
+            currentLocale = Locale.forLanguageTag(localeString);
+        }
+
+        return new BytesRef(inputByteRef.utf8ToString().toLowerCase(currentLocale));
+    }
+
+    @Override
+    public FunctionInfo info() {
+        return info;
+    }
+
+    @Override
+    public Symbol normalizeSymbol(Function symbol) {
+        assert symbol != null;
+        assert symbol.arguments().size() > 0 && symbol.arguments().size() < 3 : "invalid number of arguments";
+
+        return evaluateIfLiterals(this, symbol);
```

Comment:
this method is little misleading as it should only be used if no compilation is implemented at the scalar impl.
if compilation makes sense, like in these lower/upper functions, please implement a normalizeSymbol method which does check the optional 2nd argument. in contrast, the evaluate function should not be expected to check it, because this should be implemented at compile()